src/utils.ts

```diff
@@ -189,6 +189,19 @@ export function filterOptions(options: AnyOptions, names: string[]): AnyOptions
   return filterOptions;
 }
 
+/** @internal */
+export function disallowOptions(options: AnyOptions, denylist: string[]): AnyOptions {
+  const accumulatedOptions: AnyOptions = {};
```

Comment:
Do we worry about the performance cost of an extra iteration over the keys and also creating a new object here? I like the idea of not mutating input as much as anyone, but have we considered deleting the keys or throwing an error on the forbidden options? I think ideally we'd want to create a new set of options from the user-provided settings exactly once, and then modify them as needed; as it happens, the only place we use this helper - the get more operation - uses the object generated via destructuring in the AbstractOperation definition, so we are already working with a generated list of options.

Have we considered replacing the use of this helper in the getMore operation with a simple `delete options.comment` (or, again, just throwing a validation error)?

Reply:
Yeah, I agree in general, hence the point about making sure we create the new set of options exactly once. I think we should ensure that the top level input options are never mutated, but use the most performant way of manipulating the options as they are passed around internally. What I was saying though is that we are safe in getMore because the `super` class also spreads the options, so the input options in `AbstractCursor` get spread before they are passed to `GetMoreOperation`, and then those shallow copied options are spread *again* in the `AbstractOperation` constructor, and this helper then makes yet another copy of the copy of a copy.

Reply:
`AbstractOperation` does copy bson options, but not other user set options.  You're correct that we're safe here because we spread in `AbstractCursor` but it seems like we could easily forget that the copying is necessary.  I could write a regression test to prevent this but I think, unless there is a demonstratable performance degradation, copying is simpler.  Maybe @nbbeeken could weigh in too?

Reply:
Removed in favor of deleting `options.comment`